covenant/expected_interfaces.go

```diff
@@ -0,0 +1,44 @@
+package covenant
+
+import (
+	asig "github.com/babylonlabs-io/babylon/crypto/schnorr-adaptor-signature"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	secp "github.com/decred/dcrd/dcrec/secp256k1/v4"
+)
+
+// Signer wrapper interface to sign messages
+type Signer interface {
+	// SignTransactions signs all the transactions from the request
+	// and returns all the signatures for Slash, Unbond and Unbonding Slash.
+	SignTransactions(req SigningRequest) (*SigningResponse, error)
+	// PubKey returns the current secp256k1 public key
+	PubKey() (*secp.PublicKey, error)
+}
+
+type SigningTxsRequest struct {
+	StakingTx                       *wire.MsgTx
+	SlashingTx                      *wire.MsgTx
+	UnbondingTx                     *wire.MsgTx
+	SlashUnbondingTx                *wire.MsgTx
+	StakingOutputIdx                uint32
+	SlashingPkScriptPath            []byte
+	StakingTxUnbondingPkScriptPath  []byte
+	UnbondingTxSlashingPkScriptPath []byte
+	FpEncKeys                       []*asig.EncryptionKey
+}
+
+type SigningRequest struct {
+	SigningTxsReqByStkTxHash map[chainhash.Hash]SigningTxsRequest
```

Comment:
tbh I would avoid putting map in the in this interface as it is not clear to me how to later best serialize such request when implementing remote signer. Probably json object with with arbitrary amount of fields where key is hash and value is an object, but this sounds pretty complex 😅

I would either:
1. Have list and contract that response must have the same order as request. 
2. Avoid batches all together  and have 
```
SignDelegationTransactions(req  SigningTxsRequest ) (* SignaturesResponse , error)
```
i.e one signing request per delegation

Reply:
done at https://github.com/babylonlabs-io/covenant-emulator/pull/20/commits/97466b0c7c0395da8d48c271df791f1aaa443de5